GVFS/GVFS.Notifications/VFSForGit.Mac/StatusMenuItem/UI/VFSStatusBarItem.m

```diff
@@ -0,0 +1,100 @@
+#import "VFSAboutWindowController.h"
+#import "VFSStatusBarItem.h"
+#import "VFSProductInfoFetcher.h"
+
+@interface VFSStatusBarItem ()
+
+@property (strong, nonnull) NSStatusItem *statusItem;
+@property (strong, nonnull) VFSAboutWindowController *aboutWindowController;
+
+@end
+
+@implementation VFSStatusBarItem
+
+- (instancetype _Nullable)initWithAboutWindowController:(VFSAboutWindowController *_Nonnull)aboutWindowController
+{
+    if (self = [super init])
+    {
+        if (!aboutWindowController)
+        {
+            self = nil;
+        }
+        else
+        {
+            _aboutWindowController = aboutWindowController;
+        }
+    }
+    
+    return self;
+}
+
+- (void)load
+{
+    self.statusItem = [[NSStatusBar systemStatusBar]
+                       statusItemWithLength:NSVariableStatusItemLength];
+    
+    [self.statusItem setHighlightMode:YES];
+    
+    [self addStatusButton];
+    [self addMenuItems];
+}
+
+- (IBAction)handleMenuClick:(id)sender
+{
+    switch (((NSButton *) sender).tag)
+    {
+        case 0:
+        {
+            [self displayAboutBox];
+            break;
+        }
+        
+        default:
+        {
+            break;
+        }
+    }
+}
+
+- (IBAction)displayAboutBox
+{
+    [[NSApplication sharedApplication] activateIgnoringOtherApps:YES];
+    [self.aboutWindowController showWindow:self];
+    [self.aboutWindowController.window makeKeyAndOrderFront:self];
+}
+
+- (NSMenu *_Nullable)getStatusMenu
+{
+    return self.statusItem.menu;
+}
+
+- (void)addStatusButton
+{
+    NSImage *image = [NSImage imageNamed:@"StatusItem"];
```

Comment:
Reminder to test both light and dark modes in MacOS.

Reply:
| light  | dark |
| ------------- | ------------- |
|<img width="46" alt="Screen Shot 2019-05-31 at 11 20 09 AM" src="https://user-images.githubusercontent.com/378580/58729500-9c892c00-83b7-11e9-8125-46d02381eec1.png"> |<img width="58" alt="Screen Shot 2019-05-31 at 11 20 31 AM" src="https://user-images.githubusercontent.com/378580/58729519-aa3eb180-83b7-11e9-8c9a-642b8c8a2bfc.png">|
|<img width="172" alt="Screen Shot 2019-05-31 at 11 21 01 AM" src="https://user-images.githubusercontent.com/378580/58729579-cb070700-83b7-11e9-9f90-e810fe14ec14.png"> |<img width="187" alt="Screen Shot 2019-05-31 at 11 20 44 AM" src="https://user-images.githubusercontent.com/378580/58729590-d22e1500-83b7-11e9-9940-78a1c20fcfc0.png">|

Reply:
The About window is the same for both the modes.